enhancements/network/openstack-dual-stack.md

```diff
@@ -0,0 +1,249 @@
+  - "@cybertron for SME review on on-premise networking"
+  - "@patrickdillon for SME review and approval on installer"
+approvers:
+  - "@mandre"
+api-approvers:
+  - "None"
+creation-date: 2023-03-13
+last-updated: 2023-03-13
+tracking-link:
+  - https://issues.redhat.com/browse/OCPBU-199
+  - https://issues.redhat.com/browse/OSASINFRA-1938
+---
+
+# OpenStack dual-stack
+
+Covers adding support of OpenShift dual-stack networking on OpenStack.
+
+## Summary
+
+Customers need support for OpenShift clusters running on OpenStack with IPv4 and IPv6.
+
+## Motivation
+
+Customers need dual-stack support to allow migration of workloads from IPv4 to IPv6
+as they are running out of IPv4 addresses. This will allow OpenShift on OpenStack to
+serve a wider variety of workloads.
+
+### User Stories
+
+* As an user of a IPI OpenShift cluster, I want to have ingress and egress working with dual-stack.
+* As an user of a IPI OpenShift cluster, I want clusterIP services to be working with dual-stack.
+
+### Goals
+
+- OpenStack VMs are configured with one dual-stack interface.
+- The IPv6 Subnets can be configured with: Stateless, SLAAC or Stateful.
+- The addresses on the VM's interface are configured with the same address
+assigned by OpenStack to the port.
+- Nodes are configured with both IPv4 and IPv6 addresses, without exposing
+additional addresses, like storage interfaces.
+- Pods are configured with both IPv4 and IPv6 addresses.
+
+### Non-Goals
+
+- Support for OpenStack endpoints configured with dual-stack.
+- Support for LoadBalancer type Services.
+- Installer (IPI) handles creation of the OpenStack Network and dual-stack OpenStack Subnet.
+- Support IPv6 primary dual-stack deployments
+
+## Proposal
+
+Customers will be able to specify the IPv4 and IPv6 Subnets
+meant to be used in the `install-config.yaml`.
+
+### Workflow Description
+
+1. The customers will configure a network (tenant or provider) with IPv4 and
+IPv6 subnets. The IPv6 address mode for the Subnet can be of any type:
+SLAAC, Stateless or Stateful. Here is a sample of network and subnets creation:
+
+```bash
+$ openstack network create dual-stack
+$ openstack subnet create --network dual-stack --subnet-range 10.0.0.0/16 ipv4-subnet
+$ openstack subnet create --network dual-stack --ipv6-address-mode slaac --ipv6-ra-mode slaac --subnet-range 2001:db8:2222:5555::/64 --ip-version 6 ipv6-subnet
+```
+
+2. The customer will create API and Ingress Ports using the network created in the
+previous step. Here is a sample of ports creation:
+
+```bash
+$ openstack port create api --network dual-stack
+$ openstack port create ingress --network dual-stack
+```
+
+3. The customer will write the `install-config.yaml` and add the details of the control plane port to the new field `portTarget`.
+
+```yaml
+platform:
+  openstack:
+    ingressVIPs: ['10.0.0.147', '2001:db8:2222:5555:f816:1111:1111:1111']
+    apiVIPs: ['10.0.0.239','2001:db8:2222:5555:f816:0000:0000:0000']
+    portTarget:
```

Comment:
Do you want to update this with your latest proposal? `portTarget` is a name from an old version IIRC